src/test/fuzz/p2p_transport_deserializer.cpp

```diff
@@ -45,3 +45,14 @@ void test_one_input(const std::vector<uint8_t>& buffer)
         }
     }
 }
+
+void test_one_input(const std::vector<uint8_t>& buffer)
+{
+    std::unique_ptr<TransportDeserializer> v1_deserializer = MakeUnique<V1TransportDeserializer>(Params().MessageStart(), SER_NETWORK, INIT_PROTO_VERSION);
+    const CPrivKey k1(32, 0);
+    const CPrivKey k2(32, 0);
+    const uint256 session_id;
+    std::unique_ptr<TransportDeserializer> v2_deserializer = MakeUnique<V2TransportDeserializer>(V2TransportDeserializer(k1, k2, session_id));
+    test_deserializer(v1_deserializer, buffer, CMessageHeader::HEADER_SIZE);
+    test_deserializer(v2_deserializer, buffer, CHACHA20_POLY1305_AEAD_AAD_LEN + CHACHA20_POLY1305_AEAD_TAG_LEN);
```

Comment:
Can you explain the fuzzer a bit more? It seems you are reusing the buffer for both deserializers. Assuming that the structure of the buffer for both deserializers should be different, this might cause the fuzz engine to only focus on one of them (the "easier" one) and rarely explore paths in the v2 one, no?

If so, maybe you can split them up in two targets. See `./src/test/fuzz/deserialize.cpp` on how to do that.

Reply:
Thanks!
I switched to the same preprocessor bridge then used in `deserialize.cpp`.